components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -322,45 +333,59 @@ else if (ConfigurationFacade.getInstance().isDumbMode() && authenticatedIdPs.isE
                     }
 
                     doAuthentication(request, response, context, authenticatorConfig);
+                    /* If an authentication flow handler is redirected with incomplete, it will redirect to
+                    multi option page, as multi-option is available */
+                    if (context.getParameter(FrameworkConstants.REDIRECTED_AUTHENTICATOR) ==
+                            authenticatorConfig.getApplicationAuthenticator().getName()
+                            && isAuthenticationFlowHandler) {
```

Comment:
```suggestion
                    if (request.getAttribute(FrameworkConstants.RequestParams.FLOW_STATUS) ==
                            AuthenticatorFlowStatus.INCOMPLETE
                            && isAuthenticationFlowHandlerInMultiStep) {
```

Can't we use this? with this, we don't need to add another property.

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3533/commits/0005f836e531618f6c8b0c701cf6733750181ae3